File: frameworks/python/litestar/Implementation/routes/sse.py
from collections.abc import AsyncIterator

import msgspec
from litestar import Router, get
from litestar.response import ServerSentEvent

from payloads import Payloads


def router(p: Payloads) -> Router:
    """sse: items.medium's rows as server-sent events, through Litestar's ServerSentEvent. It writes
    what it is given as each event's data and serialises nothing itself, so each row is encoded by
    msgspec, as the json rows' are."""
    encode = msgspec.json.Encoder().encode

    async def events() -> AsyncIterator[bytes]:
        for row in p.medium.items:
            yield encode(row)

    @get("/sse/medium")
    async def medium() -> ServerSentEvent:
        return ServerSentEvent(events())

    return Router(path="/", route_handlers=[medium])

File: frameworks/node/hono/Implementation/routes/sse.ts
// rb:wiring sse.*
import { streamSSE } from "hono/streaming";

import type { Routes } from "../app.ts";

/**
 * sse: items.medium's rows as server-sent events, through streamSSE from Hono's streaming helper.
 * It writes each event to the response as it comes and ends the response when the callback returns.
 */
const sse: Routes = (app, p) => {
  app.get("/sse/medium", (c) =>
    streamSSE(c, async (events) => {
      for (const row of p.medium.items) await events.writeSSE({ data: JSON.stringify(row) });
    }));
};

export default sse;
